Added Writer field in Client and HostClient.

dgrr opened the pull request:
This commit was built to make easier the download of files.
You don't have to store the files in memory and dump in a file after that.

Solves https://github.com/erikdubbelboer/fasthttp/issues/14

dgrr committed:
Fixed problem with User-Agent header value. Added UA test in client

dgrr committed:
Deleted isGet bool var.

When client has two http request the first is GET and the second is POST it causes an error.
The only solution is reset the request structure.
With this commit I try to solve this error.

dgrr committed:
Added Writer field in Client and HostClient structures.

This commit was built to make easier the download of files.
You don't have to store the files in memory and dump in a file after that.

dgrr committed:
Aded Makefile and README

dgrr committed:
Fixed error setting Content-Length parameter in copyBody function

erikdubbelboer commented on http.go:
```diff
@@ -995,6 +995,34 @@ func (resp *Response) ReadLimitBody(r *bufio.Reader, maxBodySize int) error {
+		if n = resp.Header.ContentLength(); maxBodySize > 0 && n > maxBodySize {
+			n = maxBodySize
+		}
+		t, err = io.CopyN(wr, rd, int64(n))
```
I don't think this is going to work. Above `r` is a `bufio.Reader` which reads from `r` in blocks and buffers it in memory. So when reading the headers from `r` it might already read part of the body as well as a buffer. On this line you are reading from `rd` again which then won't include that part of the body anymore. I think this needs to read from `r` as well for this to always work.

erikdubbelboer replied:
Also this would only work with a `Content-Length` set. For Chunked or Identity encoding this would just return the encoded body instead of decoding it like it's supposed to, see [readBody](https://github.com/erikdubbelboer/fasthttp/blob/d014e440c1557aa83ecdab57d7144245c650e0f0/http.go#L1581-L1584).

dgrr replied:
Okay. I will fix and push it later.

erikdubbelboer commented on examples/writer/Makefile:
```diff
@@ -0,0 +1,6 @@
+	go build
+
+clean:
+	rm -f writer
```
Please remove the Makefile, it's not needed for such a simple example.

erikdubbelboer commented:
Can you please rebase on master so all the changes from the other commits aren't included in yours.

dgrr committed:
Added tls.Config in server structure.

This operation allow programmer to handle multiple domains with multiple certs.

dgrr committed:
Fixed error with reader
